ts/features/itwallet/trustmark/machine/machine.ts

```diff
@@ -0,0 +1,155 @@
+import { differenceInSeconds, isPast } from "date-fns";
+import { assign, fromPromise, setup } from "xstate";
+import { ItwTags } from "../../machine/tags";
+import {
+  GetCredentialTrustmarkUrlActorInput,
+  GetCredentialTrustmarkUrlActorOutput,
+  GetWalletAttestationActorOutput
+} from "./actors";
+import { Context } from "./context";
+import { Input } from "./input";
+
+const notImplemented = () => {
+  throw new Error("Not implemented");
+};
+
+export const itwTrustmarkMachine = setup({
+  types: {
+    context: {} as Context,
+    input: {} as Input,
+    events: {} as { type: "" }
+  },
+  actions: {
+    onInit: notImplemented,
+    handleSessionExpired: notImplemented,
+    updateExpirationSeconds: assign(({ context }) => ({
+      expirationSeconds: context.expirationDate
+        ? differenceInSeconds(context.expirationDate, new Date())
+        : undefined
+    })),
+    resetTrustmark: assign({
+      trustmarkUrl: undefined,
+      expirationDate: undefined,
+      expirationSeconds: undefined
+    })
+  },
+  actors: {
+    getWalletAttestationActor:
+      fromPromise<GetWalletAttestationActorOutput>(notImplemented),
+    getCredentialTrustmarkActor: fromPromise<
+      GetCredentialTrustmarkUrlActorOutput,
+      GetCredentialTrustmarkUrlActorInput
+    >(notImplemented)
+  },
+  guards: {
+    isTrustmarkExpired: ({ context }) =>
+      context.expirationDate ? isPast(context.expirationDate) : true,
+    isSessionExpired: notImplemented,
+    hasValidWalletInstanceAttestation: notImplemented
+  }
+}).createMachine({
+  id: "itwTrustmarkMachine",
+  context: ({ input }) => ({
+    credentialType: input.credentialType
+  }),
+  entry: "onInit",
+  initial: "CheckingWalletInstanceAttestation",
+  states: {
+    CheckingWalletInstanceAttestation: {
+      tags: [ItwTags.Loading],
+      description:
+        "This is a state with the only purpose of checking the WIA and decide weather to get a new one or not",
+      always: [
+        {
+          guard: "hasValidWalletInstanceAttestation",
+          target: "RefreshingTrustmark"
+        },
+        {
+          target: "ObtainingWalletInstanceAttestation"
+        }
+      ]
+    },
+    ObtainingWalletInstanceAttestation: {
+      description:
+        "This state obtains the wallet instance attestation and stores it in the context for later use in the issuance flow.",
+      tags: [ItwTags.Loading],
+      invoke: {
+        src: "getWalletAttestationActor",
+        onDone: {
+          target: "RefreshingTrustmark",
+          actions: [
```

Comment:
Do we want to store the newly obtained wallet attestation in the Redux store as well?

Reply:
Yes, the store action is right after the obtainment in the `getWalletAttestationActor` function.